homeassistant/components/avea/light.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from random import randint
+import voluptuous as vol
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ATTR_HS_COLOR, ATTR_RGB_COLOR, SUPPORT_BRIGHTNESS, SUPPORT_COLOR, Light, PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.color as color_util
+
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_AVEA = (SUPPORT_BRIGHTNESS | SUPPORT_COLOR)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Avea platform."""
+    import avea
+
+    nearbyBulbs = avea.discover_avea_bulbs()
+
+    add_devices(AveaLight(bulb) for bulb in nearbyBulbs)
+
+class AveaLight(Light):
+    """Representation of an Avea."""
+
+    def __init__(self, light):
+        """Initialize an AveaLigh."""
+        self._light = light
+        self._name = light.get_name()
```

Comment:
This is an I/O call, right?

The problem with this will be that if for some reason the device is not reachable the setup will fail.

Get all data in `setup_platform`, and if something fails due to an exception do `raise PlatformNotReady` to indicate to the HA core that it will be re-tried later.

Reply:
Like that 89b69cc6a11b6a655ebb1b01e9d379e07bd6c1c2? Sorry if this is complete bullshit, I've a complete newbie to python an ha development